data-mongodb/build.gradle

```diff
@@ -40,8 +40,14 @@ dependencies {
     testAnnotationProcessor mn.micronaut.inject.java
     testAnnotationProcessor libs.jakarta.persistence.api
 
-    testImplementation mnMongo.mongo.driver
-    testImplementation mnMongo.mongo.reactive
+    testImplementation (mnMongo.mongo.driver) {
```

Comment:
@dstepanov It seems that with latest mongodb driver 4.10.0 they introduced `KotlinCodecProvider` that returns `org.bson.codecs.kotlinx.KotlinSerializerCodecProvider` if found on class path. In this method
```
public <T> Serializer<? super T> findSerializer(Argument<? extends T> type) throws SerdeException {
        Codec<? extends T> codec = codecRegistry.get(type.getType(), codecRegistry);
        if (codec instanceof MappedCodec) {
            return ((MappedCodec<T>) codec).serializer;
        }
        if (codec != null && !(codec instanceof IterableCodec) && !(Map.class.isAssignableFrom(codec.getEncoderClass()))) {
            return new CodecBsonDecoder<>((Codec<T>) codec);
        }
        return parent.findSerializer(type);
    }
```
with new mongodb driver, codec returned is `KotlinSerializerCodec` and it fails to encode/decode field of type `CharSequence` with the error
> io.micronaut.data.exceptions.DataAccessException: Error executing PERSIST: Class 'String' is not registered for polymorphic serialization in the scope of 'CharSequence'.
To be registered automatically, class 'String' has to be '@Serializable', and the base class 'CharSequence' has to be sealed and '@Serializable'.
Alternatively, register the serializer for 'String' explicitly in a corresponding SerializersModule.

With version 4.9.1 no codec is returned for `CharSequence` and then we return our serializer `return parent.findSerializer(type);` which works fine.
Workaround for this is to exclude new mongo kotlin serializers since these are matched (although not Kotlin data classes are being encoded/decoded) and are failing with this weird error. This may be bug in mongodb driver, but our workaround is valid I think for now. There are no docs how to handle or treat these Kotlin modules/serializers. Or we shouldn't use 4.10.0 version at all.